lib/src/tracker/trackers/tracking_guard.dart

```diff
@@ -0,0 +1,32 @@
+import 'package:meta/meta.dart';
+
+import '../../events/abstract_event.dart';
+import '../abstract_tracker.dart';
+
+/// A [TrackingGuard] only forwards events
+/// to its child if shouldTrack evaluates truthy.
+@immutable
+class TrackingGuard extends AbstractTracker {
+  final AbstractTracker _child;
+  final Future<bool> Function(AbstractEvent) _shouldTrack;
+
+  /// Default initialiser
+  const TrackingGuard({
+    required AbstractTracker child,
+    required Future<bool> Function(AbstractEvent) shouldTrack,
+  })   : _child = child,
+        _shouldTrack = shouldTrack;
```

Comment:
Minor nit: I prefer using the `this.` syntax for populating properties in the constructor. It means the properties need to be public, but that seems fine here, they are final anyway.

Reply:
Guess it's just something that stuck with me from Swift development, where it is good practice to keep all fields private, unless they are 'needed' publicly. This avoids misuse as library users won't be able to circumvent the guard by calling tracker.child.track(..) directly.

Reply:
Sounds good, yeah I get that. If there is a decent reason to I am fine to keep.